File: 080/cpp/main.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int cnt = 1;
        int i = 1;
        for (int j = 1; j < nums.size(); j++) {
            if (nums[j] == nums[i - 1]) {
                if (cnt == 1) {
                    nums[i++] = nums[j];
                    cnt++;
                } else {
                    // pass
                }
            } else {
                nums[i++] = nums[j];
                cnt = 1;
            }
        }
        return i;
    }
};

File: 080/rust/main.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut i = 1;
        let mut cnt = 1;
        for j in 1..nums.len() {
            if nums[j] == nums[i - 1] {
                if cnt == 1 {
                    nums[i] = nums[j];
                    i += 1;
                    cnt += 1;
                } else {
                    // pass
                }
            } else {
                nums[i] = nums[j];
                i += 1;
                cnt = 1;
            }
        }
        return i as i32;
    }
}
